### DISCUSSION TITLE
System.AccessViolationException

### DISCUSSION THREAD

**Discussion by kuolung1** [emgucv/emgucv#971] [General]
處理序已終止，因為有未處理的例外狀況。例外狀況資訊: System.AccessViolationException 於 Emgu.CV.MatInvoke.cveMatRelease(IntPtr ByRef) 於 Emgu.CV.Mat.DisposeObject() 於 Emgu.Util.DisposableObject.Dispose()

**Comment by emgucv** [comment]
Please recreate the issue with a few lines of code and share it here. 

We will look into that if we can re-produce the issue on our side.

**Comment by kuolung1** [comment]
In my case, after the program was executed for a long time (about 5 to 8 hours), the program crashed for unknown reasons. I only saw this line in the windows even log, and there was no indication where it was executed. Moreover, in my program, Even after deleting all the Mat.Dispose() code, this situation still occurs. Check the reason. It may be that when the system executes GC, the memory in mat is recycled, but the needDispose in the EmguCV record is not there. It was caused by clearing and executing Dispose again, but because my code does not directly execute Dispose(), I guess there is a bug in the automatic execution of Dispose() in EmguCV. The possible relevant codes are as follows :

`
public void SetImage(Mat image) {
    if (image != null) {
        lock (ImageLock) {
            if (SetImageCount > SetImageCountLimit) {
                if (Dimage != null) {
                    Dimage = null;
                    Dimage = new Mat(new Size(image.Width, image.Height), Emgu.CV.CvEnum.DepthType.Cv8U, 1);
                } else {
                    Dimage = new Mat(new Size(image.Width, image.Height), Emgu.CV.CvEnum.DepthType.Cv8U, 1); }
                SetImageCount = 0;
            } else {
                if (Dimage == null) {
                    Dimage = new Mat(new Size(image.Width, image.Height), Emgu.CV.CvEnum.DepthType.Cv8U, 1); }
                SetImageCount++; } }
        if (bwithMaskImage) {
            double beta = 1.0 - alpha;
            CvInvoke.AddWeighted(image, alpha, _MaskImage, beta, 0.0, Dimage);
        } else{
            image.CopyTo(Dimage); }
        ImageInput(Dimage);
        DrawImage();
        DisplayImage();
        GC.KeepAlive(Dimage); }}
`

This program captures multiple industrial cameras and displays them on the screen

ps : the code segment a little problem

**Comment by emgucv** [comment]
The code fragment is not compilable. Please submit a few (<20) lines of compilable code that can reproduce the issue and we will be able to take a look.

**Comment by kuolung1** [comment]
public void SetImage(Mat image)
        {
            if (image != null)
            {
                lock (ImageLock)
                {
                    if (SetImageCount > SetImageCountLimit)
                    {
                        if (Dimage != null)
                        {
                            Dimage = null;
                            Dimage = new Mat(new Size(image.Width, image.Height), Emgu.CV.CvEnum.DepthType.Cv8U, 1);
                        }
                        else
                        {
                            Dimage = new Mat(new Size(image.Width, image.Height), Emgu.CV.CvEnum.DepthType.Cv8U, 1);
                        }
                        SetImageCount = 0;
                    }
                    else
                    {
                        if (Dimage == null)
                        {
                            Dimage = new Mat(new Size(image.Width, image.Height), Emgu.CV.CvEnum.DepthType.Cv8U, 1);
                        }
                        SetImageCount++;
                    }
                }

                if (bwithMaskImage)
                {
                    double beta = 1.0 - alpha;
                    CvInvoke.AddWeighted(image, alpha, _MaskImage, beta, 0.0, Dimage);
                } else
                {
                    image.CopyTo(Dimage);
                }

                ImageInput(Dimage);
                DrawImage();
                DisplayImage();

                GC.KeepAlive(Dimage);
            }
        }

**Reply by emgucv** [reply]
The ImageInput(), DrawImage(), DisplayImage() functions are all not available, alone with many variables in your code. We won't be able to run it. Please strip off all none-related functions and variables, make the code self-contained for us to run.
